tests/test_resource_upload.py

```diff
@@ -255,135 +119,40 @@ def test_upload_resources_no_existing_metadata_file(practice_filesystem,
-                             )
-    stubber.add_response('put_object', {},
-                         expected_params={
-                            'Body': botocore.stub.ANY,
-                            'Bucket': bucket_name,
-                            'Key': full_key3,
-                            'ContentType': 'application/json'
-                            }
-                         )
-    stubber.activate()
-    mocker.patch('boto3.client', lambda service: s3_client)
-
-    resource_dir = practice_filesystem[test_dir]
-
-    metadata_path = practice_filesystem[metadata_file]
-    csv_path = practice_filesystem[csv_file]
-    mocker.patch(
-        "sys.argv",
-        ["", resource_dir, metadata_path, bucket_name,
-         s3_dir, "--csv", csv_path]
-    )
-    copy_resources_s3.main()
-
-    expected_csv_data = [
-        {
-            "filename": "example1.txt",
-            "hash": "dc330ae2bc1d0b2edac442ed3f8245647cf5c0c0"
-        },
-        {
-            "filename": "example3.jpeg",
-            "hash": "e606bc6acc83666e1d40722e9c743a01e12e65ab"
-        },
-        {
-            "filename": "example2.mp4",
-            "hash": "a31e7f061d762f1e5099ecebfe6877310e5be420"
-        },
-        {
-            "filename": "example4.jpeg",
-            "hash": "e606bc6acc83666e1d40722e9c743a01e12e65ab"
+    stubber.assert_no_pending_responses()
+
+    with open(index_path, 'r') as f:
+        file_data = f.read()
+    soup = BeautifulSoup(file_data, "html.parser")
+    table = soup.find("table", class_="table table-striped table-bordered")
+    rows = table.find_all("tr")
+    sha_data_map_from_index = {}
+
+    for row in rows:
+        cells = row.find_all("td")
+        data = [cell for cell in cells]
+        # first element is empty because <tr> uses <th> instead of <td>.
+        if data == []:
+            continue
+
+        # test prefix
+        assert os.path.dirname(data[2].text.strip()) == url_prefix
+
+        sha_data_map_from_index[data[2].text.strip().split("/")[-1]] = {
+            'mime_type': data[1].text.strip(),
+            'path': data[2].text.strip()
         }
-    ]
-
-    with open(csv_path, 'r') as f:
-        reader = csv.DictReader(f)
-        data = [row for row in reader]
-        assert len(data) == 4
-        for row in data:
-            assert row in expected_csv_data
-
-
-def test_add_metadata_indent(tmp_path):
-    metadata_file = tmp_path / "media.json"
-    metadata_file.write_text("[]")
-
-    copy_resources_s3.add_metadata(
-        [{"foo": 1, "bar": 2}],
-        str(metadata_file)
-    )
-
-    text = metadata_file.read_text()
 
-    assert text == """[
-  {
-    "foo": 1,
-    "bar": 2
-  }
-]"""
+    # Make sure each table row corresponds with the correct key, path, and
+    # mime_type in sha_map
+    for key, value in sha1_map.items():
+        sha_data_map_from_index[key].get('path') == value['path']
+        sha_data_map_from_index[key].get('mime_type') == value['mime_type']
```

Comment:
I think you meant to add `assert` here, but they are missing. Adding them does cause test failures.

I'm a bit unsure as to what these assertions are doing. I think currently the test case is actually missing a bug it should have caught in that the output is missing an expected file. We should probably have the test case assert that a link exists for every file in `practice_filesystem` :man_shrugging: .